### ISSUE TITLE
No such file or directory "Wifi.h and WiFiClientSecure.h"

### ISSUE THREAD

**Issue by MohammedNoureldin** [Links2004/arduinoWebSockets#635]
I am trying to add WebSocket to my ESP32 project, but unfortunately the compiler fails. Following is the error message I get:

```
.pio/libdeps/esp32_dev/WebSockets/src/WebSockets.h:190:30: fatal error: WiFiClientSecure.h: No such file or directory
.pio/libdeps/esp32_dev/WebSockets/src/WebSockets.h:189:18: fatal error: WiFi.h: No such file or directory
```

I am using PlatformIO IDE, below is my `platformio.ini` (the important part):

```
[env:esp32_dev]
framework = arduino
platform_packages = 
	; Uses new tool chain with a non-released Arduino package to be able to compile with C++17 (C++17 may become the default in the future, therefore this may be soon not necessary).
	toolchain-xtensa@~5.100200.0
	framework-arduinoespressif32@https://github.com/espressif/arduino-esp32.git
build_unflags =
	-std=c++11
	-fno-exceptions ; Removes the default exceptions disabling flag.
build_flags =
	-std=c++17
	-D SERIAL_BAUD_RATE=${common.monitor_speed}
	-D NO_GLOBAL_SERIAL1
platform = espressif32
board = esp32dev
lib_deps =
        ...
	links2004/WebSockets @ ^2.3.5
```

Could anyone please help me to get it compiling? I am not sure what is wrong, because my project compiles as long as I do not add WebSocket library to it.

**Comment by Links2004** [comment]
this is normal and is based on how platform io is working.

you have to ways to get the "Wifi.h"
auto detect via:
```
lib_ldf_mode = chain+
```
or manual via:
```
lib_deps = 
	WiFi
        WiFiClientSecure
```

**Comment by MohammedNoureldin** [comment]
@Links2004 Hi Markus, thanks for your reply. I am a bit confused. I am already using `chain+`, and this is exactly what triggers the problem, please find the minimal project I provided here:

https://github.com/platformio/platformio-core/issues/3876#issuecomment-799509343

**Comment by Links2004** [comment]
ok, I where able to reproduce after some updates.
but I dont understand why this happens 
for the ESP8266 `lib_ldf_mode = chain+` is working fine,
but with the ESP32 it failes.

**Comment by MohammedNoureldin** [comment]
@Links2004 , exactly, that was my observation as well. Do you have any idea what to do? I already reported this as a bug in PlatformIO repo.

https://github.com/platformio/platformio-core/issues/3876#issuecomment-799509343

**Comment by MohammedNoureldin** [comment]
Hi Markus, do you think it makes sense to build something temporary into the code or `library.json` as kind of workaround until PlatformIO guys fix the bug?

Please check their temporary workaround here:

https://github.com/platformio/platformio-core/issues/3876#issuecomment-800163580

**Comment by Links2004** [comment]
the problem is that `chain+` is required for ESP8266 and the `NO_GLOBAL_INSTANCES` define see https://github.com/Links2004/arduinoWebSockets/issues/624 
forcing `chain` for the lib is no option since it will brake other code bases.

**Comment by MohammedNoureldin** [comment]
@Links2004 that is funny that the other issue was opened by meو too. I will think a bit about it then let you know if anything useful comes to my mind. Thank you Markus!

**Comment by lemonkey** [comment]
ugh

**Comment by ggoksel** [comment]
Thank you links2004..

**Comment by yoroboticscolombia** [comment]
HI, did you get it fixed?

**Comment by steefst33f** [comment]
for me actually removing:

`lib_ldf_mode = deep+`

in the platform.ini file made it compile without any errors.
No idea why though...

**Comment by bbuster2** [comment]
Any update on this?
I am having the same errors with an ESP32. I need 
`lib_ldf_mode = deep+` or 
`lib_ldf_mode = chain+`, otherwise it won't compile because BLEDevice.h is missing etc.

Even adding lib_deps= WiFi and WiFiClientSecure gives me the same errors
